source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -135,6 +135,15 @@ To select `configuration.nix` in the working directory, specify the configuratio
 nix-build '<nixpkgs/nixos>' -A vm -I nixos-config=./configuration.nix
 ```
 
+:::{note}
+On systems that do not have a `$NIX_PATH` environment variable set up as recommended in [the pinning tutorial](pinning-nixpkgs) you use:
+
+```shell-session
+nix-build $(nix-prefetch-url --print-path --unpack https://github.com/NixOS/nixpkgs/archive/nixos-22.11.tar.gz | tail -n 1)/nixos -A vm -I nixos-config=./configuration.nix
+
+nix-prefetcch-url downloads the archive of Nixpkgs, unpacks the archive, and returns the nix store path. This path is then used for by nix-build command to build the virtual machine.
+```
+:::
```

Comment:
I don't know. No, not really. I was just messing around to figure out if it can be done. This is not practical at all (if only because of re-downloading) and just distracts from the purpose. But what we can reasonably do is make the search path explicit using another search path entry. It's a bit messy still, but that's how things are in Nix 2.x...

```console
nix-build '<nixpkgs/nixos>' -A vm -I https://github.com/NixOS/nixpkgs/tarball/78342239aac6382af25986219ffb163ba79b155a -I nixos-config=./configuration.nix
```

And possibly break it down into multiple lines so it's easier to grasp:


```console
nix-build '<nixpkgs/nixos>' -A vm \
-I https://github.com/NixOS/nixpkgs/tarball/78342239aac6382af25986219ffb163ba79b155a \
-I nixos-config=./configuration.nix
```